### PR TITLE
LDC93S1 Simple sample importer

### LINKED ISSUE #72: Revive LDC93S1
Right now, running codebase for end-to-end execution without any interest in the actual training of a real network is at best dependant on TED-LIUM dataset: 34GB download, 40GB disk space.

As far as I can tell, this is non usable for anyone wanting quick iterations on CPU or small GPU. Worst, I have killed good systems several times because of how much memory it eats (Core i7 4790K, 32GB of RAM, 45GB of Swap and GTX970).

Let's re-introduce a simple LDC93S1-based sample for that kind of usecase.

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #72

### CODE REVIEW

**Comment by lissyx** [comment]
This needs to be merged after issue #75 is merged, but it should be good to review already @kdavis-mozilla

File: util/importers/ldc93s1.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    _ = base.maybe_download(LDC93S1_BASE + ".txt", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".txt")
+
+    # Conditionally extract LDC93S1 data
+    LDC93S1_DIR = "LDC93S1"
+    _maybe_extract(data_dir, LDC93S1_DIR, local_file, ["dev", "test", "train"])
+
+    # Create dev DataSet
+    train = dev = test = _read_data_set(graph, data_dir, LDC93S1_DIR, "dev", thread_count, batch_size, numcep, numcontext)
+
+    # Create test DataSet
```
Comment (kdavis-mozilla) [inline util/importers/ldc93s1.py]: Could you remove these commented out lines or un-comment them and have the line above only assign to `dev`

File: util/importers/ldc93s1.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    LDC93S1_BASE_URL = "https://catalog.ldc.upenn.edu/desc/addenda/"
+    local_file = base.maybe_download(LDC93S1_BASE + ".wav", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".wav")
+    _ = base.maybe_download(LDC93S1_BASE + ".txt", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".txt")
+
+    # Conditionally extract LDC93S1 data
+    LDC93S1_DIR = "LDC93S1"
+    _maybe_extract(data_dir, LDC93S1_DIR, local_file, ["dev", "test", "train"])
+
+    # Create dev DataSet
+    train = dev = test = _read_data_set(graph, data_dir, LDC93S1_DIR, "dev", thread_count, batch_size, numcep, numcontext)
```
Comment (kdavis-mozilla) [inline util/importers/ldc93s1.py]: I understand all the data sets are the same, but `_read_data_set()` is taking a `"dev"`argument which confuses things.

File: util/importers/ldc93s1.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    LDC93S1_BASE = "LDC93S1"
+    LDC93S1_BASE_URL = "https://catalog.ldc.upenn.edu/desc/addenda/"
+    local_file = base.maybe_download(LDC93S1_BASE + ".wav", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".wav")
+    _ = base.maybe_download(LDC93S1_BASE + ".txt", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".txt")
+
+    # Conditionally extract LDC93S1 data
+    LDC93S1_DIR = "LDC93S1"
+    _maybe_extract(data_dir, LDC93S1_DIR, local_file, ["dev", "test", "train"])
+
+    # Create dev DataSet
```
Comment (kdavis-mozilla) [inline util/importers/ldc93s1.py]: Comment is talking about old code.

File: util/importers/ldc93s1.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    # Create wav dir
+    wav_dir = path.join(data_dir, extracted_data, data_set)
+
+    # Obtain list of txt files
+    txt_files = glob(path.join(wav_dir, "*.txt"))
+
+    # Let us generate one batch for each thread that is going to run
+    txt_files = [ txt_files[0] for x in xrange(batch_size * 2) ]
+
+    print("I CAN HAZ txt_files=", len(txt_files), txt_files)
```
Comment (kdavis-mozilla) [inline util/importers/ldc93s1.py]: Remove

File: util/importers/ldc93s1.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+def _read_data_set(graph, data_dir, extracted_data, data_set, thread_count, batch_size, numcep, numcontext):
+    # Create wav dir
+    wav_dir = path.join(data_dir, extracted_data, data_set)
+
+    # Obtain list of txt files
+    txt_files = glob(path.join(wav_dir, "*.txt"))
+
+    # Let us generate one batch for each thread that is going to run
+    txt_files = [ txt_files[0] for x in xrange(batch_size * 2) ]
```
Comment (kdavis-mozilla) [inline util/importers/ldc93s1.py]: To me this makes no sense.

In the method `_create_files_circular_list()` the threads are enumerating over
```
self._files_circular_list
```
which is a [circular list](https://docs.python.org/2.7/library/itertools.html#itertools.cycle) in which each item is the same. So, this serves no purpose.

File: run-ldc93s1.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
Comment (lissyx) [inline run-ldc93s1.sh]: done

File: util/importers/ldc93s1.py (hunk 3 above)
Comment (lissyx) [inline util/importers/ldc93s1.py]: removed that

File: util/importers/ldc93s1.py (hunk 2 above)
Comment (lissyx) [inline util/importers/ldc93s1.py]: cleaned this  by removing the whole notion of dev/train/test subdirs

File: util/importers/ldc93s1.py (hunk 1 above)
Comment (lissyx) [inline util/importers/ldc93s1.py]: done

File: util/importers/ldc93s1.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        available_gpus = get_available_gpus()
+        return max(len(available_gpus), 1)
+
+    def _start_queue_threads(self):
+        batch_threads = [Thread(target=self._populate_batch_queue) for i in xrange(self._thread_count)]
+        for batch_thread in batch_threads:
+            batch_thread.daemon = True
+            batch_thread.start()
+
+    def _create_files_circular_list(self):
```
Comment (lissyx) [inline util/importers/ldc93s1.py]: killed that

File: util/importers/ldc93s1.py [hunk 8]
```diff
@@ -0,0 +1,155 @@
+import wave
```
Comment (lissyx) [inline util/importers/ldc93s1.py]: :)

File: util/importers/ldc93s1.py [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+          wav_file = path.splitext(txt_file)[0] + ".wav"
+          wav_file_size = getsize(wav_file)
+          priorityQueue.put((wav_file_size, (txt_file, wav_file)))
+        files_list = []
+        while not priorityQueue.empty():
+            priority, (txt_file, wav_file) = priorityQueue.get()
+            files_list.append((txt_file, wav_file))
+        return cycle(files_list)
+
+    def _populate_batch_queue(self):
```
Comment (lissyx) [inline util/importers/ldc93s1.py]: That should be good and I could do something simpler, and that looks to be running as expected :)

**Comment by kdavis-mozilla** [comment]
LGTM

**Pull request merged**